File: 563-BinaryTreeTilt/C++/main.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
private:
    int result = 0;
    
public:
    int helper(TreeNode* root) {        
        int left = 0, right = 0;
        if (root->left) {
            left = helper(root->left);
        }
        
        if (root->right) {
            right = helper(root->right);
        }
        
        result += abs(left - right);
        return left + right + root->val;
    }
    int findTilt(TreeNode* root) {
        if (!root) return 0;
        
        helper(root);
        return result;
    }
};

File: 563-BinaryTreeTilt/Python/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def findTilt(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        self.res = 0
        
        def helper(root):           
            left, right = 0, 0
            if root.left:
                left = helper(root.left)
            
            if root.right:
                right = helper(root.right)
            
            self.res += abs(left - right)
            
            return left + right + root.val
        
        if not root:
            return 0
        
        helper(root)
        return self.res